visualization/app/codeCharta/ui/codeMap/threeViewer/threeOrbitControlsService.ts

```diff
@@ -56,30 +103,52 @@ export class ThreeOrbitControlsService {
 		this.threeCameraService.camera.translateZ(oldZoom)
 	}
 
-	public autoFitTo(obj = this.threeSceneService.mapGeometry) {
-		const boundingSphere = new THREE.Box3().setFromObject(obj).getBoundingSphere()
+	private setDefaultZoom() {
+		this.defaultZoom = this.getZoom()
+	}
 
+	private autoFitToCalculation(boundingSphere) {
 		const cameraReference = this.threeCameraService.camera
 
 		const scale = 1.4 // object size / display size
 		const objectAngularSize = ((cameraReference.fov * Math.PI) / 180) * scale
 		const distanceToCamera = boundingSphere.radius / Math.tan(objectAngularSize / 2)
 		const len = Math.sqrt(Math.pow(distanceToCamera, 2) + Math.pow(distanceToCamera, 2))
 
-		cameraReference.position.set(len, len, len)
-		this.controls.update()
+		return len
+	}
 
+	private focusCameraViewToCenter(boundingSphere) {
 		const t: Vector3 = boundingSphere.center.clone()
 		t.setY(0)
 		this.controls.target.set(t.x, t.y, t.z)
 
-		cameraReference.lookAt(t)
-
-		this.defaultCameraPosition = cameraReference.clone().position
+		this.threeCameraService.camera.lookAt(t)
 
 		this.threeCameraService.camera.updateProjectionMatrix()
 	}
 
+	private initializeDefaultZoomWithoutAutoFit() {
+		const autoFittedPerspective: Vector3 = new Vector3(0, 0, 0)
+		const boundingSphere = this.getBoundingSphere()
+		const len: number = this.autoFitToCalculation(boundingSphere)
+
+		autoFittedPerspective.set(len, len, len)
+
+		this.defaultCameraPosition = autoFittedPerspective.clone()
+
+		const t: Vector3 = boundingSphere.center
+		t.setY(0)
+
+		const centerVector: Vector3 = new Vector3(t.x, t.y, t.z)
+
+		this.defaultZoom = autoFittedPerspective.distanceTo(centerVector)
+	}
+
+	private getBoundingSphere() {
```

Comment:
Could it be that the bounding box here is not correct for focused nodes? (The new issue)